slog_test.go

```diff
@@ -0,0 +1,71 @@
+//go:build go1.21
+
+/*
+Copyright 2023 The logr Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logr_test
+
+import (
+	"errors"
+	"fmt"
+	"log/slog"
+	"os"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/logr/funcr"
+)
+
+var debugWithoutTime = &slog.HandlerOptions{
+	ReplaceAttr: func(groups []string, a slog.Attr) slog.Attr {
+		if a.Key == "time" {
+			return slog.Attr{}
+		}
+		return a
+	},
+	Level: slog.LevelDebug,
+}
+
+func ExampleFromSlog() {
+	logger := logr.FromSlog(slog.New(slog.NewTextHandler(os.Stdout, debugWithoutTime)))
+
+	logger.Info("hello world")
+	logger.Error(errors.New("fake error"), "ignore me")
+	logger.WithValues("x", 1, "y", 2).WithValues("str", "abc").WithName("foo").WithName("bar").V(4).Info("with values, verbosity and name")
+
+	// Output:
+	// level=INFO msg="hello world"
+	// level=ERROR msg="ignore me" err="fake error"
+	// level=DEBUG msg="foo/bar: with values, verbosity and name" x=1 y=2 str=abc
+}
+
+func ExampleToSlog() {
+	logger := logr.ToSlog(funcr.New(func(prefix, args string) {
```

Comment:
I was thinking about adding slog support to funcr, but decided against it because:
- here I want to test the code in slogHandler, which wouldn't be used if funcr itself supported slog
- the handlers that come with slog provide similar functionality, so I don't see the need

Reply:
However, klog and zapr definitely should get extended. I've started doing that for zapr, using the same interface for conversion that I am proposing here.

Reply:
For zapr see https://github.com/go-logr/zapr/pull/60

Reply:
Because `logr.ToSlog` and `log.FromSlog` are using it to avoid glue code. `slogr.New` and `slogr.NewHandler` (your PR) don't, but [I think they should](https://github.com/go-logr/logr/pull/196#issuecomment-1665051819) if we go with that API.